pylearn2/datasets/svhn.py

```diff
@@ -46,44 +61,48 @@ class SVHN(dense_design_matrix.DenseDesignMatrixPyTables):
 
     def __init__(self, which_set, path=None, center=False, scale=False,
                  start=None, stop=None, axes=('b', 0, 1, 'c'),
-                 preprocessor = None):
+                 preprocessor = None, enable_cache=True,
+                 write_permission=False):
 
         assert which_set in self.mapper.keys()
 
         self.__dict__.update(locals())
         del self.self
 
-        if path is None:
-            path = '${PYLEARN2_DATA_PATH}/SVHN/format2/'
+        if write_permission and (not enable_cache):
+            if path is None:
+                raise ValueError("You are not allowed to change the original" +
+                                 " file located at $PYLEARN2_DATA_PATH. " +
+                                 " Enable caching to remove this error")
+            else:
+                warnings.warn("You might change the data in the files " +
+                              "located at: {}".format(path))
```

Comment:
Under what circumstances? Can you be more specific? Will this class ever try and alter them?